manifests/mod/passenger.pp

```diff
@@ -1,36 +1,527 @@
+  $rails_app_spawner_idle_time                                                               = undef,
+  $rails_auto_detect                                                                         = undef,
+  $rails_autodetect                                                                          = undef,
+  $rails_base_uri                                                                            = undef,
+  $rails_default_user                                                                        = undef,
+  $rails_env                                                                                 = undef,
+  $rails_framework_spawner_idle_time                                                         = undef,
+  $rails_ruby                                                                                = undef,
+  $rails_spawn_method                                                                        = undef,
+  $rails_user_switching                                                                      = undef,
+  $union_station_filter                                                                      = undef,
+  $union_station_gateway_address                                                             = undef,
+  $union_station_gateway_cert                                                                = undef,
+  $union_station_gateway_port                                                                = undef,
+  $union_station_key                                                                         = undef,
+  $union_station_proxy_address                                                               = undef,
+  $union_station_support                                                                     = undef,
+  $wsgi_auto_detect                                                                          = undef,
 ) inherits ::apache::params {
-
   include ::apache
-
+  if $passenger_installed_version {
+    if $passenger_allow_encoded_slashes {
+      if (versioncmp($passenger_installed_version, '4.0.0') < 0) {
+        fail("Passenger config option :: passenger_allow_encoded_slashes is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
+      }
+    }
+    if $passenger_app_env {
+      if (versioncmp($passenger_installed_version, '4.0.0') < 0) {
+        fail("Passenger config option :: passenger_app_env is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
+      }
+    }
+    if $passenger_app_group_name {
+      if (versioncmp($passenger_installed_version, '4.0.0') < 0) {
+        fail("Passenger config option :: passenger_app_group_name is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
+      }
+    }
+    if $passenger_app_root {
+      if (versioncmp($passenger_installed_version, '4.0.0') < 0) {
```

Comment:
Would it be easier to sort the parameters based on their version instead of duplicating the versioncmp logic a bunch?
```puppet
    if (versioncmp($passenger_installed_version, '4.0.0') < 0) {
      if $passenger_allow_encoded_slashes {
        fail("Passenger config option :: passenger_allow_encoded_slashes is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
      }
      if $passenger_app_env {
        fail("Passenger config option :: passenger_app_env is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
      }
      if $passenger_app_group_name {
        fail("Passenger config option :: passenger_app_group_name is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
      }
      if $passenger_app_root {
        fail("Passenger config option :: passenger_app_root is not introduced until version 4.0.0 :: ${passenger_installed_version} is the version reported")
      }
    }

```

Reply:
Normally we keep these alphabetized just to make things easier to find. If sorting by version in this case works better, then I'm fine with it.